src/cdk/overlay/position/cover-position-strategy.ts

```diff
@@ -0,0 +1,608 @@
+
+/**
+ * A strategy for positioning overlays. Using this strategy, an overlay is given an
+ * implicit position relative some origin element. The relative position is defined in terms of
+ * a point on the origin element that is connected to a point on the overlay element. For example,
+ * a basic dropdown is connecting the bottom-left corner of the origin to the top-left corner
+ * of the overlay.
+ */
+export class CoverPositionStrategy implements PositionStrategy {
+  /** The overlay to which this strategy is attached. */
+  private _overlayRef: OverlayReference;
+
+  /** Whether we're performing the very first positioning of the overlay. */
+  private _isInitialRender: boolean;
+
+  /** Whether the overlay can be pushed on-screen on the initial open. */
+  private _canPush = true;
+
+  /** Whether the overlay can grow via flexible width/height after the initial open. */
+  private _growAfterOpen = false;
+
+  /** Whether the overlay's width and height can be constrained to fit within the viewport. */
+  private _hasFlexibleDimensions = true;
+
+  /** Whether the overlay position is locked. */
+  private _positionLocked = false;
+
+  /** Amount of space that must be maintained between the overlay and the edge of the viewport. */
+  private _viewportMargin = 0;
+
+  /** Cached viewport dimensions. */
+  private _viewportRect: BoundingBoxRect;
+
+  /**
+   * The origin element against which the horizontal start of the overlay will be positioned.
+   * The start of the overlay will match the start of this element.
+   */
+  private _startOrigin?: FlexibleConnectedPositionStrategyOrigin;
+
+  /**
+   * The origin element against which the horizontal end of the overlay will be positioned.
+   * The end of the overlay will match the end of this element.
+   * If absent, the end of _startOrigin will be used.
+   */
+  private _endOrigin?: FlexibleConnectedPositionStrategyOrigin;
+
+  /**
+   * The origin element against which the top of the overlay will be positioned.
+   * The top of the overlay will match the top of this element.
+   */
+  private _topOrigin?: FlexibleConnectedPositionStrategyOrigin;
+
+  /**
+   * The origin element against which the bottom of the overlay will be positioned.
+   * The bottom of the overlay will match the bottom of this element.
+   * If absent, _topOrigin will be used.
+   */
+  private _bottomOrigin?: FlexibleConnectedPositionStrategyOrigin;
+
+  /** The overlay pane element. */
+  private _pane: HTMLElement;
+
+  /** Whether the strategy has been disposed of already. */
+  private _isDisposed: boolean;
+
+  /**
+   * Parent element for the overlay panel used to constrain the overlay panel's size to fit
+   * within the viewport.
+   */
+  private _boundingBox: HTMLElement | null;
+
+  /** The last computed bounding box. */
+  private _lastBoundingBox: Partial<BoundingBoxRect>|null;
+
+  /** The last computed bounding box adjustments. */
+  private _lastAdjustments: BoundingBox|null;
```

Comment:
It's not fully clear from the name what "adjustments" means. I would expect something like `_last____Offset`, where the blank is whatever the offset is necessary for (like `lastScrollOffset` or `lastResizeOffset` etc.)